lib/airtable/entity/table.rb

```diff
@@ -6,9 +7,9 @@ class Table
       DEFAULT_DIRECTION = 'asc'.freeze
 
       def initialize(base, name)
-        @name = name
+        @name = CGI.escape(name)
```

Comment:
I don't think CGI.escape works - we need to use an equivalent of encodeUriComponent in JS . 

https://github.com/Airtable/airtable-ruby/commit/3cc21eaf699be2eec2c0c135c042f61508cff878

For example they differ in handling of spaces - GCI.escape turns them into '+' v.s. %2B.

Separately, storing @name as encoded might be confusing later on. If someone does `p table.name`, they would n't expect the encoding. It'd be better to either:
a) have an accessor url_encoded_name that does encoding on the fly.
b) wrap name access in url_encode e.g. `@table_path = [@base, url_encode(@name)]`
c) add in instance variable `@url_encoded_name = url_encode(@name)`